tests/cancun/eip1153_tstore/test_tload_calls.py

```diff
@@ -0,0 +1,114 @@
+"""
+Ethereum Transient Storage EIP Tests
+https://eips.ethereum.org/EIPS/eip-1153
+"""
+
+from typing import Dict, Union
+
+import pytest
+
+from ethereum_test_tools import (
+    Account,
+    Address,
+    Environment,
+    StateTestFiller,
+    TestAddress,
+    Transaction,
+)
+from ethereum_test_tools.vm.opcode import Bytecode
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-1153.md"
+REFERENCE_SPEC_VERSION = "2f8299df31bb8173618901a03a8366a3183479b0"
+
+
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize("call_type", [Op.CALL, Op.CALLCODE, Op.DELEGATECALL])
+def test_tload_calls(state_test: StateTestFiller, call_type: Op):
+    """
+    Ported .json vectors:
+
+    (04_tloadAfterCallFiller.yml)
+    Loading a slot after a call to another contract is 0.
+
+    (12_tloadDelegateCallFiller.yml)
+    delegatecall reads transient storage in the context of the current address
+    """
+    address_to = Address("A00000000000000000000000000000000000000A")
+    address_call = Address("B00000000000000000000000000000000000000B")
+
+    # Storage variables
+    slot_a_tload_after_subcall_result = 0
+    slot_a_subcall_result = 1
+    slot_b_subcall_tload_result = 2
+    slot_b_subcall_updated_tload_result = 3
+
+    def make_call(call_type: Op) -> Bytecode:
+        if call_type == Op.DELEGATECALL or call_type == Op.STATICCALL:
+            return call_type(Op.GAS(), address_call, 0, 32, 0, 0)
+        else:
+            return call_type(Op.GAS(), address_call, 0, 0, 32, 0, 0)
```

Comment:
We really should have a macro for this. `Macro.ANYCALL` or something that automatically formats the inputs. More important now that we also have the EOF `EXT*CALL` opcodes.

I can try to implement this in the same PR that we do the parametrization marker.